include/oneapi/dpl/experimental/kt/esimd_radix_sort_utils.h

```diff
@@ -66,14 +66,28 @@ gather(const T* input, sycl::ext::intel::esimd::simd<::std::uint32_t, N> offsets
     return sycl::ext::intel::esimd::gather(input + base_offset, offsets * size32<T>, mask);
 }
 
-template <typename T, int N, sycl::access_mode Mode, sycl::access::placeholder P>
+template <typename T, int N, sycl::access_mode Mode, sycl::access::placeholder P, ::std::enable_if_t<sizeof(T) <= sizeof(::std::uint32_t), int> = 0>
 sycl::ext::intel::esimd::simd<T, N>
 gather(sycl::accessor<T, 1, Mode, sycl::target::device, P> input, sycl::ext::intel::esimd::simd<::std::uint32_t, N> offsets,
        ::std::uint32_t base_offset, sycl::ext::intel::esimd::simd_mask<N> mask = 1)
 {
     return sycl::ext::intel::esimd::gather<T>(input, offsets * size32<T>, base_offset * size32<T>, mask);
 }
 
+template <typename T, int N, sycl::access_mode Mode, sycl::access::placeholder P, ::std::enable_if_t<sizeof(T) == sizeof(::std::uint64_t), int> = 0>
+sycl::ext::intel::esimd::simd<T, N>
+gather(sycl::accessor<T, 1, Mode, sycl::target::device, P> input,
```

Comment:
I suggest that we get rid of `utils::gather` and `utils::scatter` functions in favor of using `lsc_gather` and `lsc_scatter` directly, as it already done in cooperative and onesweep implementations.

LSC counterparts do not need abstractions due to having compatible API for accessors and pointers.

Reply:
I am not sure that we should do these changes.
As for me better to use ```utils::gather``` and ```utils::scatter``` functions in all radix sort implementations.

Reply:
Probably, adding abstractions to avoid adding things like `uint32_t(sizeof(KeyT))` worth, because conversion to uint32_t is not straightforward.